src/components/selectionBoxLayer.ts

```diff
@@ -83,11 +81,53 @@ export module Components {
         topLeft: topLeft,
         bottomRight: bottomRight
       };
-      this._bindBoxDataValues();
+    }
+
+    // sets pixelPosition() or value() based on the other, depending on which was the last one set
+    private _syncPixelPositionAndValue() {
+      if (this._xScale != null) {
+        if (this._xBoundsMode === PropertyMode.VALUE && this._xExtent[0] != null
+            && this._xExtent[1] != null) {
+          let topLeft = {
+            x: this._xScale.scale(this._xExtent[0]),
+            y: this.bounds().topLeft.y
+          };
+          let bottomRight = {
+            x: this._xScale.scale(this._xExtent[1]),
+            y: this.bounds().bottomRight.y
+          };
+          this._setBounds({
+            topLeft: topLeft,
+            bottomRight: bottomRight
+          });
+        } else if (this._xBoundsMode === PropertyMode.PIXEL) {
+          this._xExtent = [this._xScale.invert(this.bounds().topLeft.x), this._xScale.invert(this.bounds().bottomRight.x)];
```

Comment:
But the extent can always be calculated by inverting the box bounds, as long as there is a scale. Keeping things in sync is bad.